### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Eldenroot**
Change cache rebuild, add language files, remove some queries.

Thank you @supryk!

### CODE REVIEW

**Comment by Eldenroot** [comment]
@lairdshaw - Hi buddy, please could you review? Thank you!

File: UPLOAD/inc/plugins/purgesoftdeleted.php [hunk 1]
```diff
@@ -63,8 +67,8 @@ function purgesoftdeleted_activate()
 				
 				// If not then add
 					$new_task = array(
-						"title" => "Purge soft deleted posts and threads",
-						"description" => "Checks for soft deleted posts and threads and purges them automatically.",
+						"title" => $db->escape_string($lang->purgesoftdeleted_task_name),
+						"description" => $db->escape_string($lang->purgesoftdeleted_task_desc)
```
Comment (lairdshaw) [inline UPLOAD/inc/plugins/purgesoftdeleted.php]: A comma is missing at the end of this line.

**Comment by Eldenroot** [comment]
@lairdshaw - thank you, comma is added :)

**Review by lairdshaw — APPROVED** [review]
Looks good to me.

**Comment by Eldenroot** [comment]
@lairdshaw - I re-indented the files---should be better and fix #5

File: UPLOAD/inc/languages/english/admin/config_purgesoftdeleted.lang.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+$l['purgesoftdeleted_name'] = "Purge soft deleted posts and threads";
+$l['purgesoftdeleted_desc'] = "Automatically purges soft deleted posts and threads";
+
+$l['purgesoftdeleted_task_name'] = "Purge soft deleted posts and threads";
+$l['purgesoftdeleted_task_desc'] = "Checks for soft deleted posts and threads and purges them automatically.";
+
+$l['purgesoftdeleted_task_log'] = "Purge soft deleted posts and threads now are pernamenly deleted";
```
Comment (lairdshaw) [inline UPLOAD/inc/languages/english/admin/config_purgesoftdeleted.lang.php]: Typo: "pernamenly" => "permanently". Also, we can probably strip the "Purge" from the start of this message, and replace it with an indication that age matters, for something like:

"Older soft deleted posts and threads were permanently deleted"

File: UPLOAD/inc/languages/english/purgesoftdeleted.lang.php [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<?php
+
+$l['purgesoftdeleted_task_log'] = "Purge soft deleted posts and threads now are pernamenly deleted";
```
Comment (lairdshaw) [inline UPLOAD/inc/languages/english/purgesoftdeleted.lang.php]: Is this necessary given that it's already in the admin language file?

File: UPLOAD/inc/languages/english/admin/config_purgesoftdeleted.lang.php (hunk 2 above)
Comment (Eldenroot) [reply UPLOAD/inc/languages/english/admin/config_purgesoftdeleted.lang.php]: Fixed, thx!

File: UPLOAD/inc/languages/english/purgesoftdeleted.lang.php (hunk 3 above)
Comment (Eldenroot) [reply UPLOAD/inc/languages/english/purgesoftdeleted.lang.php]: Fixed, thx!

**Pull request merged**